core/src/main/java/com/linecorp/armeria/server/HttpServerUpgradeHandler.java

```diff
@@ -75,8 +79,15 @@ final class HttpServerUpgradeHandler extends ChannelInboundHandlerAdapter {
 
     private static final FullHttpResponse UPGRADE_RESPONSE = newUpgradeResponse();
 
+    private static final ByteBuf INVALID_SETTINGS_HEADER_MESSAGE = Unpooled.unreleasableBuffer(
+            Unpooled.directBuffer()
+                    .writeBytes("Invalid HTTP2-Settings header\n".getBytes(StandardCharsets.UTF_8)));
+
     private static final Splitter COMMA_SPLITTER = Splitter.on(',').omitEmptyStrings().trimResults();
 
+    private static final FullHttpResponse invalidSettingsHeaderResponse =
+            new DefaultFullHttpResponse(HTTP_1_1, BAD_REQUEST, INVALID_SETTINGS_HEADER_MESSAGE.duplicate());
```

Comment:
No need to duplicate here because we always duplicate below

Reply:
Let me push a commit for that. 😉